### ISSUE TITLE
Setting header-parameters in API calls

### ISSUE THREAD

**Issue by kmelve** [engram-design/FeedMe#31]
It would be handy having the possibility to set header-parameters and authentication/OAuth in the API calls. e.g [Placid](https://github.com/alecritson/Placid/blob/master/placid/services/Placid_OAuthService.php) has it. I solved it by writing a plugin between PLACID and Feedme, it works, but it would be nice to have this out of the box.

**Comment by engram-design** [comment]
Yep, its on our [roadmap](https://github.com/engram-design/FeedMe#roadmap) of things to do soon!

**Comment by siebird** [comment]
@kmelve looking to do something similar as most of the the feeds I'm accessing need authentication. Do you have that plugin shared publicly?

**Comment by engram-design** [comment]
@siebird Its not quite at Oauth level yet, but you can use `curlOptions` in your config file to setup auth for curl - via http://sgroup.com.au/plugins/feedme/getting-started/configuration

**Comment by kmelve** [comment]
@siebird I'm afraid not, as it's custom made for a client of ours. I just make use the [Placid plugin service](https://github.com/momentumhq/Placid/wiki/Create-request-in-your-plugin) in my own plugin controller and hook FeedMe through the action-url. I'm not sure I do it the same way today, but it works, and gives the client some configurability in the control panel. It's probably easier to just set `curlOptions` as @engram-design advices.

**Comment by RichardFrontwise** [comment]
For craft3: create a feed-me.php file in /config where you can specify clientOptions and requestOptions to setup custom authentication

**Comment by pieter-janDB** [comment]
`As for OAuth, I'm tracking that in a new FR` This link is no longer available.. Is it yet possible to import from google sheets API using OAuth ? (I know it's already possible with apikey) ? Maybe in combination with the `OAuth 2.0 Client` plugin?
